psydac/api/fem.py

```diff
@@ -544,12 +665,19 @@ def allocate_matrices(self, backend=None):
                 if self._matrix:
                     global_mats[0,0] = self._matrix
                 else:
-                    global_mats[0,0] = StencilMatrix(trial_space, test_space, pads=tuple(pads), backend=backend)
+                    global_mats[0,0] = StencilMatrix(trial_space, test_space, pads=tuple(pads))
 
                 md                 = global_mats[0,0].domain.shifts
                 mc                 = global_mats[0,0].codomain.shifts
                 diag               = compute_diag_len(pads, md, mc)
                 self._matrix       = global_mats[0,0]
+
+        if backend is not None and is_broken:
+            for mat in global_mats.values():
+                mat.set_backend(backend)
+        elif backend is not None:
+            self._matrix.set_backend(backend)
+
         return  global_mats.values()
```

Comment:
Instead of returning this list, you should just set `self._global_matrices`.

Please also add a proper docstring to this function.

Reply:
done in [ec10ae7](https://github.com/pyccel/psydac/pull/213/commits/ec10ae70e2106c6cc339a5aa2ae5ac81c351a7f6) and [faee8ae](https://github.com/pyccel/psydac/pull/213/commits/faee8aee6c7650f331121b939f604925c9dd86db)